cg/meta/upload/fohm/fohm.py

```diff
@@ -0,0 +1,199 @@
+
+    @staticmethod
+    def create_joined_dataframe(file_list: List[Path]) -> pd.DataFrame:
+        """Creates datafame with all csv files used in daily delivery"""
+        dataframe_list = [pd.read_csv(filename, index_col=None, header=0) for filename in file_list]
+        return pd.concat(dataframe_list, axis=0, ignore_index=True)
+
+    def append_metadata_to_aggregation_df(self) -> None:
+        """
+        Add field with internal_id to dataframe
+        """
+
+        self.aggregation_dataframe["internal_id"] = self.aggregation_dataframe["provnummer"].apply(
+            lambda x: self.status_db.samples_by_ids(name=x).first().internal_id
+        )
+        self.aggregation_dataframe["region_lab"] = self.aggregation_dataframe["internal_id"].apply(
+            lambda x: f"{self.lims_api.get_sample_attribute(lims_id=x, key='region_code').split(' ')[0]}"
+            f"_{self.lims_api.get_sample_attribute(lims_id=x, key='lab_code').split(' ')[0]}"
+        )
+
+    def link_sample_rawdata(self) -> None:
+        """
+        Hardlink samples rawdata files to fohm delivery folder
+        """
+        for sample_id in self.aggregation_dataframe["internal_id"]:
+            sample_obj: models.Sample = self.status_db.sample(sample_id)
+            bundle_name = sample_obj.links[0].family.internal_id
+            version_obj: Version = self.housekeeper_api.last_version(bundle=bundle_name)
+            files = self.housekeeper_api.files(version=version_obj.id, tags=[sample_id]).all()
+            for file in files:
+                shutil.copy(file.full_path, Path(self.daily_rawdata_path))
+
+    def reaggregate_reports(self):
+        unique_regionlabs = list(self.aggregation_dataframe["region_lab"].unique())
+        for region_lab in unique_regionlabs:
+            pangolin_df = self.pangolin_dataframe[
+                self.aggregation_dataframe["region_lab"] == region_lab
+            ]
+            pangolin_df.to_csv(
+                self.daily_rawdata_path
+                / f"{region_lab}_{self.current_datestr}_pangolin_classification_format3.txt",
+                sep="\t",
+                index=False,
+            )
+            report_df = self.reports_dataframe[
+                self.aggregation_dataframe["region_lab"] == region_lab
+            ]
+            report_df.to_csv(
+                self.daily_report_path / f"{region_lab}_{self.current_datestr}_komplettering.csv",
+                sep=",",
+                index=False,
+            )
+
+    def assemble_fohm_delivery(self, cases: List[str]) -> None:
+        """Rearrange data and put in delivery dir"""
+        self._cases_to_aggregate = cases
+        self.create_daily_delivery_folders()
+        self.append_metadata_to_aggregation_df()
+        self.reaggregate_reports()
+        self.link_sample_rawdata()
+
+    def send_mail_reports(self, date: Optional[str] = None):
+        if date:
+            self._current_datestr = date
```

Comment:
Why is this here? I do not see that it is used and why mutate `self._current_datestr`then?